### DISCUSSION TITLE
Resolving {cfr} blocker dependencies

### DISCUSSION THREAD

**Discussion by pratikunterwegs** [epiverse-trace/epiverse-trace.github.io#85] [General]
This discussion is aimed at moving {cfr} closer to an initial release on CRAN.

{cfr} relies on two dependencies that are not on CRAN, {epiparameter} and {covidregionaldata}. {epiparameter} is imported and is key to how {cfr} handles delay distributions, while {covidregionaldata} is only suggested and is used in the vignettes (and was used in some examples, but see draft PR 58 in the {cfr} repo).

{cfr} cannot be released to CRAN until these two dependencies are resolved, i.e., either removed or put on CRAN themselves.

## {covidregionaldata}

{covidregionaldata} could be removed and a snapshot of the UK covid data used from {incidence2} instead - but this would disallow a very nice cross-country comparison in one of the vignettes. This is relevant since we have identified LMIC rather than UK examples as something we want to include more of.

One alternative would be to download some of that data and contribute a snapshot of it to another package such as {outbreaks}. However, {oubreaks}' Readme still says "the purpose of the current package is only to share examplar datasets during the hackathon" and the package was last updated on CRAN in 2020. So this might not be a quick solution - does anybody have an estimate of how this might go?

A second alternative would be to include this data within {cfr} itself, as it already includes an ebola dataset.

In any case, it would be good to hear thoughts from @Bisaloo or @sbfnk as among those in Epiverse involved with {covidregionaldata}.

## {epiparameter}

This is a key dependency, so unlikely to be removed - is there any chance we could get a timeline for when it might get to CRAN? I'd be happy to pitch in if some focused work is required to make progress on that. Pinging @adamkucharski and @joshwlambert for thoughts.

**Comment by TimTaylor** [comment]
Minor comment. IMO - the {outbreaks} README needs updating as I think the "hackathon" part is a remnant and misleading. AFAIK the package is aiming to be a more centralised provider for example data. I'm not the maintainer though so could you raise an issue to clarify?

**Reply by pratikunterwegs** [reply]
Thanks Tim - appreciate the Readme might be a bit out of date. Let's see how this discussion pans out and if the consensus is to contribute to {outbreaks} I can definitely raise an issue exploring the possibility there.

**Comment by jamesmbaazam** [comment]
> I'd be happy to pitch in if some focused work is required to make progress on that.

I am happy to help here too. This is where a typical sprint could help complete it in a few days.

**Reply by pratikunterwegs** [reply]
Indeed, might be worth discussing in/for mid Sep once the LSHTM team are all back in the office.

**Comment by jamesmbaazam** [comment]
> {cfr} cannot be released to CRAN until these two dependencies are resolved, i.e., either removed or put on CRAN themselves.

In preparation to release {bpmodels} on CRAN, I had to remove {epiparameter} as a Suggested package because it was used in the vignette. However, I plan to revert to its use once it's on CRAN.

**Reply by pratikunterwegs** [reply]
Thanks, good to know. I don't think this would be a good direction for {cfr}, so I'd rather wait (or work) to have {epiparameter} on CRAN before pushing out {cfr}. In the meantime, other checks could certainly be satisfied first.

**Reply by jamesmbaazam** [reply]
Yes, at this point, getting {epiparameter} on CRAN should be a priority as that seems to be a blocker on other packages. I imagine a similar blocker on the horizon for {superspreading} as well.

**Comment by Bisaloo** [comment]
> In any case, it would be good to hear thoughts from @Bisaloo or @sbfnk as among those in Epiverse involved with {covidregionaldata}.

It would take a huge effort to put covidregionaldata back on CRAN, and its functionality would likely be severely reduced as multiple data sources disappeared after the peak of the epidemic. I don't think it's likely to ever make it back on CRAN.

outbreaks is a good option and is already used in linelist.

A custom minimal dataset could be a good option as well.

**Reply by pratikunterwegs** [reply]
Thanks, would you have a preference between raising an issue in {outbreaks} vs the custom dataset? The first one is naturally likely to be a slower process, and we might possibly want to add or modify this data at some point, so having it in {cfr} itself could be more convenient.

A further thought, are there any licensing issues to consider with the data?

**Reply by Bisaloo** [reply]
You didn't mention using data already present in outbreaks. Is this an option?

**Reply by pratikunterwegs** [reply]
I have checked. We need both case and death data for {cfr}, and for the time-varying method, an outbreak of a substantial duration. Additionally, the specific example where {covidregionaldata} is useful is a cross-country comparison. {outbreaks} has no comparable dataset unless I've missed something, happy to be pointed towards it if so.

**Reply by Bisaloo** [reply]
In this case, I would say do both:

- start by including a snapshot of a minimal dataset in cfr to remove the blocker you mention in the opening comment
- once this is done, open an issue in outbreaks to propose the inclusion of a new dataset with a short explanation of what properties (missing from the existing datasets) make it an interesting example

This allows us to move fast and remove the current blocker while trying to improve the global ecosystem.

**Reply by adamkucharski** [reply]
Yep, could import from any alternative COVID data source that has cases and deaths over sufficiently long period to expect a change from accumulation of immunity (e.g. timeseries from OWID? https://ourworldindata.org/coronavirus). {covidregionaldata} was used as illustrative example in early versions of package, so not reason this has to be the go-to dependency.

I've recently been in touch with Finlay Campbell about adding extra datasets to {outbreaks}, rather than having them in our packages (e.g. the more extensive Ebola line list data compiled by Alicia et al: https://elifesciences.org/articles/09015) and he said package is open to PRs.

**Reply by pratikunterwegs** [reply]
Thanks both - @adamkucharski's suggestion to try {owidR} works as the `owid_covid` dataset fits our requirements well, and the package is on CRAN. I'll swap that in for {covidregionaldata} and that issue can tentatively be considered resolved.

For the second point, I'm happy to make the PR to outbreaks for the ebola linelist too, so we can coordinate on that in the next couple of weeks - I see the data are available from eLife, but something for a different discussion thread.

If we feel it's good to have, I can still take up the second step of @Bisaloo's suggestion to add some {covidregionaldata} to {outbreaks} - happy to hear opinions on that, although it is not super useful for us now.

**Comment by joshwlambert** [comment]
> ## {epiparameter} 
> This is a key dependency, so unlikely to be removed - is there any chance we could get a timeline for when it might get to CRAN?

I think having as stable {epiparameter} release to go to CRAN for the end of August is realistic. This is after the CRAN summer break. Happy to discuss and adjust is {cfr} needs to go to CRAN sooner than this.

**Reply by pratikunterwegs** [reply]
Thanks Josh, there isn't pressure from my side on meeting a specific deadline but definitely an open offer to help get {epiparameter} out. It would be good to get {cfr} out in say, mid-to-late September, but let's see what we can manage.

**Comment by pratikunterwegs** [comment]
Closing as this issue has been resolved in https://github.com/epiverse-trace/cfr/pull/88 and https://github.com/epiverse-trace/cfr/pull/58
